### ISSUE TITLE
Daily video on ios

### ISSUE THREAD

**Issue by anthodb** [daily-demos/custom-video-daily-react-hooks#26]
Hello Daily team, we are trying to make the rooms work on ios but we get an error each time it starts : 
```
Error while creating  DailyCall object
TypeError: undefined is not an object (evaluating 'navigator.mediaDevices.ondevicechange')
(fonction anonyme) — daily-iframe-esm.js:1:79874
i2 — daily-iframe-esm.js:1:84240
value — daily-iframe-esm.js:1:85271
```
Do you have any idea on the origin of the issue? Is the react video hook working on ios? Maybe you can share a working sample?

**Comment by lazeratops** [comment]
Hey @anthodb, thanks for reporting this, we'll investigate! Which iOS version are you on, and are you running Safari?

**Comment by anthodb** [comment]
Hey @lazeratops, we are working on iOs 15.0.2 with Safari. Let us know if you need other infos! Thanks

**Comment by nienkedekker** [comment]
Hey @anthodb - apologies for the delay in responding. I tried reproducing your issue, but couldn't. Are you still running into this issue and if so, where? Locally, or when deployed on Netlify? Thanks! :-)

**Comment by NTAK666** [comment]
I also have same problem, I use project with typescript, problem is not only in IOS but also in Android, is work on Web Browser

"typescript": "4.9.5",
"@daily-co/daily-js": "^0.48.0",
"@daily-co/daily-react": "^0.10.0",
"react": "18.2.0",

**Comment by nienkedekker** [comment]
Hey @NTAK666, so you're saying the demo works on Android/iOS in a web browser - in what other context doesn't it work? The more info, the better! Thank you :)

**Comment by NTAK666** [comment]
> Hey @NTAK666, so you're saying the demo works on Android/iOS in a web browser - in what other context doesn't it work? The more info, the better! Thank you :)

No, I mean it works in Desktop web browsers and in Android/iOS web browsers (Safari/Chorme) it doesn't work

**Comment by nienkedekker** [comment]
Thanks! I can replicate the issue on Safari on MacOS and Safari on iOS (I currently don't have an Android phone to test on, unfortunately). First question: are you testing a video call "Safari to Safari" (i.e. opening the same video call in two different tabs)? 

If so, this looks like a “standard” Safari testing issue :( Using the same Safari profile in both tabs means only one video will work. If you're doing this, try installing the [Safari Technology Preview](https://developer.apple.com/safari/technology-preview/) and then use both Safari’s to do the Safari-to-Safari test. You want to make sure that you use 2 different browser profiles for testing.

Let me know if this doesn't solve the issue and I can take another look on Monday.

**Comment by NTAK666** [comment]
> Thanks! I can replicate the issue on Safari on MacOS and Safari on iOS (I currently don't have an Android phone to test on, unfortunately). First question: are you testing a video call "Safari to Safari" (i.e. opening the same video call in two different tabs)?
> 
> If so, this looks like a “standard” Safari testing issue :( Using the same Safari profile in both tabs means only one video will work. If you're doing this, try installing the [Safari Technology Preview](https://developer.apple.com/safari/technology-preview/) and then use both Safari’s to do the Safari-to-Safari test. You want to make sure that you use 2 different browser profiles for testing.
> 
> Let me know if this doesn't solve the issue and I can take another look on Monday.

Thank for your time @nienkedekker, I think you just found a new issue.

  
1\. About my problem: I think my issue is not there because each time I only use 1 device (Iphone) iOS or Android, it can come from typescript when checking the data type. At that time \`mediaDevices\` was not received so it got type undefined?

2\. The problem you mention: I think it is related to the available devices on your computer, for example my computer (Windows 11) can use 2 tabs at the same time, but when opening the 3rd tab, it has problem.

My project is quite complicated so I can't deploy it yet. Please feel free to contact me if you need more information.

**Comment by nienkedekker** [comment]
Hm, weird. Can you post the error here, and the exact repro steps to trigger the error? Thanks!
